grc/gui_qt/components/canvas/block.py

```diff
@@ -443,6 +443,15 @@ def mousePressEvent(self, e):
 
         self.moveToTop()
 
+    def contextMenuEvent(self, e):
+        if not self.isSelected():
+            self.parent.clearSelection()
+            self.setSelected(True)
+
+        view = self.parent.view
+        contextMenu = view._app().MainWindow.menus["edit"]
+        pos = view.mapFromScene(e.pos())
```

Comment:
`e.pos()` is in block-relative coordinates, so you will have to map to scene coords before you can map to view coords